RFC-0001-Agile Coretime.md

```diff
@@ -0,0 +1,298 @@
+
+## Design
+
+### Overview
+
+Upon implementation of this proposal, slot auctions and associated crowdloans cease. Instead, Coretime on the Polkadot UC is sold by the Polkadot System in two separate formats: *Bulk* and *Instantaneous*. This proposal only mandates the implementation of *Bulk Coretime*; any mentions of *Instantaneous Coretime* is only intended to illustrate a possible final solution.
+
+*Bulk Coretime* is sold periodically and allocated any time in advance of its usage, whereas *Instantaneous Coretime* is sold immediately prior to usage on a block-by-block basis with an explicit allocation at the time of purchase.
+
+All Bulk Coretime sold by Polkadot is done so on a new system parachain known as the *Broker-chain*. Revenue from sales would be sent to the Polkadot Treasury. Owners of Bulk Coretime are tracked on this chain and the ownership status and properties (i.e. the specific period) of the owned Coretime are exposed over XCM as a non-fungible asset.
+
+At the request of the owner, the Broker-chain allows Bulk Coretime to be:
+
+1. Transferred in ownership.
+2. Split into quantized, non-overlapping segments of Bulk Coretime with the same ownership.
+3. Consumed in exchange for the allocation of a Core during its period.
+4. Consumed in exchange for a pro-rata amount of the revenue from Instantaneous Core sales over its period.
+
+###  Special Considerations
+
+As a migration mechanism, pre-existing leases (from the legacy lease/slots/crowdloan framework) SHALL be recorded in the Broker-chain and cores assigned to them. When the lease comes to expiry, there is a mechanism to gain a priority sale of Coretime to ensure that the Parachain suffers no downtime.
+
+Additionally, there is a renewal system which allows a core's Para assignment to be renewed unchanged with a known maximum price increase from month to month. Note that the intention behind renewals is only to ensure that committed Paras get some guarantees about price for predicting future costs. As such this price-capped renewal system only allows cores to be reused for their same tasks from month to month. In any other context, the regular sale system must be used and the regular price paid.
+
+### Relay-chain and Instantaneous Coretime
+
+Sales of Instantaneous Coretime are expected to happen on the Polkadot Relay-chain. The Relay-chain is expected to be responsible for:
+
+- holding non-transferable, non-refundable DOT balance information for collators.
+- setting and adjusting the price of Instantaneous Coretime based on usage.
+- allowing collators to consume their DOT balance in exchange for the ability to schedule one PoV for near-immediate usage.
+- ensuring the Broker-chain has timely book-keeping information on Coretime sales revenue. This should include a total instantaneous revenue amount for each block number.
```

Comment:
why is it necessary that the broker-chain receives this information?

Reply:
To be able to make proper rewards.